frontend/src/routes/_account.plan.index.tsx

```diff
@@ -0,0 +1,80 @@
+// Copyright 2025 New Vector Ltd.
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+// Please see LICENSE in the repository root for full details.
+
+import { queryOptions, useSuspenseQuery } from "@tanstack/react-query";
+import { Navigate, createFileRoute } from "@tanstack/react-router";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { graphql, useFragment } from "../gql";
+import { graphqlRequest } from "../graphql";
+
+export const CONFIG_FRAGMENT = graphql(/* GraphQL */ `
+  fragment PlanManagement_siteConfig on SiteConfig {
+    planManagementIframeUri
+  }
+`);
+
+const QUERY = graphql(/* GraphQL */ `
+  query SiteConfig {
+    siteConfig {
+      ...PlanManagement_siteConfig
+    }
+  }
+`);
+
+export const query = queryOptions({
+  queryKey: ["siteConfig"],
+  queryFn: ({ signal }) => graphqlRequest({ query: QUERY, signal }),
+});
+
+export const Route = createFileRoute("/_account/plan/")({
+  loader: ({ context }) => context.queryClient.ensureQueryData(query),
```

Comment:
also you should `throw redirect({ to: "/" })` if there is no plan management URI here, this should allow you to avoid loading the site config in the `_account.index.tsx` and just redirect to `/plan` there?

Reply:
I've gone ahead and reverted to the simpler version with no check at the query param handling level.

Reply:
> I changed it to check preemptively based on a verbal conversation that we had... I would happily revert if you confirm that is what you prefer?

Sorry I think there was a small misunderstanding here 😅 
I think I was thinking about using `throw redirect` in the loader of the plan route, so that it never tries to display the route